src/utils/api.js

```diff
@@ -0,0 +1,147 @@
+// utils/Api.js
+
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    // constructor body
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getUserInfo(), this.getInitialCards()])
+      .then(([userInfo, cards]) => {
+        // console.log("Cards:", cards); // Debugging
+        if (!Array.isArray(cards)) {
+          console.error("Expected an array but received:", cards);
+          return;
+        }
+
+        return [userInfo, cards];
+      })
+      .catch(console.error);
+  }
+
+  getInitialCards() {
+    console.log("Fetching initial cards...");
+    return (
+      fetch(`${this._baseUrl}/cards`, {
+        headers: this._headers,
+      })
+        .then((res) => {
+          // console.log("Raw response:", res);
+          if (res.ok) {
+            return res.json();
+          }
+          return Promise.reject(`Error: ${res.status}`);
+        })
+        // .then((data) => {
+        //   // console.log("Cards received:", data);
+        //   console.log("Cards received (should be array):", Array.isArray(data));
+        //   return data;
+        // })
+        .catch((err) => console.error("API error:", err))
```

Comment:
* You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.